packages/holocron/src/hooks/useHolocron.jsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+/* eslint-disable no-underscore-dangle, react/jsx-pascal-case */
+
+import React, { useMemo } from 'react';
+import { Provider } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import useModuleRegistry from './useModuleRegistry';
+import createHolocronStore from '../createHolocronStore';
+
+export const HolocronContext = React.createContext({});
+
+export function useHolocronContext() {
+  return React.useContext(HolocronContext);
+}
+
+export default function useHolocron(options = {}) {
+  const {
+    // store config
+    store: providedStore,
+    reducer = (state) => state,
+    initialState,
+    enhancer,
+    localsForBuildInitialState,
+    extraThunkArguments,
+    // registry config
+    blockedModules,
+    holocronModules,
+    holocronModuleMap,
+  } = options;
+
+  const registry = useModuleRegistry(holocronModuleMap, holocronModules, blockedModules);
+  const [store] = React.useState(() => providedStore || createHolocronStore({
+    reducer,
+    initialState,
+    enhancer,
+    localsForBuildInitialState,
+    extraThunkArguments: { ...extraThunkArguments, registry },
+  }));
+
+  const context = useMemo(
+    () => ({ store, registry }),
+    [store, registry]
+  );
+
+  return React.useMemo(() => {
+    function __Holocron__({ children }) {
```

Comment:
Where does this private React component convention come from? Any articles for background you can share with us?